### ISSUE TITLE
Can this plugin remove the background of a game card instead of a human portrait?

### ISSUE THREAD

**Issue by Wells83** [occ-ai/obs-backgroundremoval#555]
Hi everybdoy. Its possible to use this plugin for create a virtual background on a square, for exemple a card game ?

**Comment by umireon** [comment]
@Wells83 Firstly, could you please provide a more detailed description of your desired use case along with some screenshots? The information you provide will help us in providing a more accurate answer.

**Comment by Wells83** [comment]
Yes, of course!

First of all, excuse my English, I'm French :)

Here's what I want to do, for my youtube channel, I want to inlay game cards into my main scene.

Initially, my camera will film my cards like this:

![Screenshot_2024-03-10_14-08-11](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/5c595220-3731-4482-b766-72aa05b1048b)

I then add green borders and chrominance overlay:

![Screenshot_2024-03-10_14-09-03](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/11c92c22-581f-4eb3-9ef6-bf6db1b075f6)

Unfortunately, in doing so, I lose the green in my cards :(

So I'd like to create a single-color virtual background that detects the outline of the cards, so that I can inlay them perfectly without losing the green they contain.

Importantly, when I don't have a card to show, the scene must be totally transparent.

Without card :

![Screenshot_2024-03-10_14-09-22](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/49797249-6f91-4e4b-ade7-e1001114e689)

With card :

![Screenshot_2024-03-10_14-09-18](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/6151030b-8988-41e2-9ed4-4caf0e1e5079)



Last but not least, I obviously have several card sizes, orientations and formats:

![Screenshot_2024-03-10_14-10-17](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/019833c6-6fb4-4b5c-a51c-204344564fc0)
![Screenshot_2024-03-10_14-10-31](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/801b4ebf-2b02-419e-8210-c32a4cb47ec4)
![Screenshot_2024-03-10_14-10-42](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/90e15382-02af-45bd-8f01-3174fd915c7f)


To sum up, the objective is :

![Screenshot_2024-03-10_14-08-11 - Copie](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/85f180ee-539f-4d3f-bbc5-190b00509133)

**Comment by umireon** [comment]
@royshil What do you think this suggestion?

**Comment by umireon** [comment]
@Wells83 I understand your needs as you want to remove the background of game cards based on AI. Our matt model is based on human portraits so that it cannot recognize your game cards. Is my understanding correct?

**Comment by royshil** [comment]
It makes a lot.of sense
There are "segment anything" models and background removal model that are not for humans necessarily but for any object in the foreground, which I think would work for this pretty good.

Like Bria RMBG https://huggingface.co/briaai/RMBG-1.4
Which has onnx already and is quantized which means it may run close to realtime

**Comment by umireon** [comment]
@royshil Do you mean @Wells83 's demand is easily feasible?

**Comment by royshil** [comment]
Yes I don't think it's very hard to implement, honestly since we have everything set up in the plugin already, like onnxruntime and the image & models processing utility

**Comment by umireon** [comment]
The milestone is set to 1.1.11

**Comment by Wells83** [comment]
Thinking about it, I wonder if the best thing wouldn't be for it to detect the green zone and replace it with a virtual background.

That way, if there's no game card, we're fine and if there is a card, of course it has to detect the outline of the card.

I don't know if it's the same on your side or if it's more complicated?

**Comment by umireon** [comment]
I suppose detecting the green zone and detecting the object is almost the same thing. Just the matt is inverted and it won't make any difference in the implementaion.

**Comment by Wells83** [comment]
I receive a notification when thé New version IS done ?

**Comment by umireon** [comment]
We will close this issue when we have done the implementation. We cannot promise that your suggestion will be implemented soon but stay tuned! 😄

**Comment by royshil** [comment]
implemented.

**Comment by Wells83** [comment]
Ok i test that tomorow ans i give you feedback

**Comment by Wells83** [comment]
I made a test scene:

- 1 green background image
- 1 blue image simulating the card

where should the filter be placed?

what setting should I use?

![image](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/c19acd96-429c-48a7-a54d-b7aca71a9022)

**Comment by Wells83** [comment]
If its the good parameter, its not working :(

![image](https://github.com/occ-ai/obs-backgroundremoval/assets/11405506/d0d818f5-e906-4bd6-a99f-99ec8ce260bd)

**Comment by royshil** [comment]
@Wells83 You need the RMBG model
It is only available in the 1.1.12 version. You may need to upgrade

**Comment by Wells83** [comment]
yes its the last version i use

**Comment by royshil** [comment]
@Wells83 do you see the RMBG option for the model?
